tower-util/src/ext/then.rs

```diff
@@ -10,21 +10,21 @@ pub struct Then<A, B> {
     b: B,
 }
 
-impl<A, B> Then<A, B>
-where
-    A: Service,
-    B: Service<Request = Result<A::Response, A::Error>, Error = A::Error> + Clone,
-{
+impl<A, B> Then<A, B> {
     /// Create new `Then` combinator
-    pub fn new(a: A, b: B) -> Then<A, B> {
+    pub fn new<Request>(a: A, b: B) -> Then<A, B>
+    where
+        A: Service<Request>,
+        B: Service<Result<A::Response, A::Error>, Error = A::Error> + Clone,
+    {
         Then { a, b }
     }
 }
 
 impl<A, B> Clone for Then<A, B>
```

Comment:
This might be an unrelated change, but this seems like it should be derive-able?

Reply:
Yeah, it probably should be fixed in another PR.